internal/reconcilers/messages/messages.go

```diff
@@ -22,21 +22,28 @@ import (
 
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/google/uuid"
+
+	"github.com/stacklok/minder/internal/entities/properties"
 )
 
 // RepoReconcilerEvent is an event that is sent to the reconciler topic
 type RepoReconcilerEvent struct {
 	// Project is the project that the event is relevant to
 	Project uuid.UUID `json:"project"`
-	// Repository is the repository to be reconciled
-	Repository int64 `json:"repository" validate:"gte=0"`
+	// Provider is the provider that the event is relevant to
+	Provider uuid.UUID `json:"provider"`
+	// Properties is the properties of the repository to be reconciled
+	Properties map[string]any `json:"properties"`
 }
 
 // NewRepoReconcilerMessage creates a new repos init event
 func NewRepoReconcilerMessage(providerID uuid.UUID, repoID int64, projectID uuid.UUID) (*message.Message, error) {
```

Comment:
With these changes and especially when issue #4618 is implemented, we can just pass in properties and use this to reconcile a generic entity.